### PR TITLE
Hc6 profile

### PR DESCRIPTION
**Pull request by Oleksii30**

### CODE REVIEW

File: frontend/src/app.tsx [hunk 1]
```diff
@@ -3,11 +3,13 @@ import { Route, Switch } from 'react-router-dom';
 import { AppRoute } from 'common/enums';
 import Sign from 'components/sign/sign'
 import NotFound from 'components/not-found/not-found';
+import Profile from 'components/profile/profile';
 import DoctorsSearch from 'components/doctors-search/doctors-search';
 
 const App: React.FC = () => (
   <Switch>
     <Route path={[AppRoute.SIGN_IN, AppRoute.SIGN_UP]} component={Sign}/>
+    <Route path="/profile" exact component={Profile} />
```
Comment (ghost) [inline frontend/src/app.tsx]: Use `AppRoute` enum here

File: frontend/src/components/profile/components/profile-info/profile-info.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import React from "react";
+import { IUser } from 'common/interfaces/user';
+import { getDateFormat } from 'helpers';
+import clsx from "clsx";
+import styles from './styles.module.scss';
+
+type Props = {
+  user: IUser | null
+  edit: () => void
+}
```
Comment (ghost) [inline frontend/src/components/profile/components/profile-info/profile-info.tsx]:
```suggestion
type Props = {
  user: IUser;
  edit: () => void
}
```
use on<Element><Event> for component props cb

File: frontend/src/components/profile/components/profile-info/profile-info.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+const ProfileInfo: React.FC<Props> = ({user, edit}) => {
+  let birthdate
+  if (user) birthdate = getDateFormat(user?.birthdate)
```
Comment (ghost) [inline frontend/src/components/profile/components/profile-info/profile-info.tsx]: Always user curly

Comment (ghost) [inline frontend/src/components/profile/components/profile-info/profile-info.tsx]: Create helper for it.
Something like `const getFormattedDate = (date, formattedDate: FormattedDateType) => {}`

File: frontend/src/components/profile/components/profile-info/profile-info.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          <span className={clsx(styles.icon, styles.pen)}></span>
+        </div>
+      </div>
+      <div className={styles.infoBloks}>
+        <div
+          className={styles.photo}
+          style={{
+            backgroundImage: `url(${user?.imagePath})`
+          }}
+        ></div>
```
Comment (ghost) [inline frontend/src/components/profile/components/profile-info/profile-info.tsx]: User image tag here

File: frontend/src/components/profile/components/profile-info/profile-info.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        <div className={styles.mainUserInfo}>
+          <div className={styles.card}>Card: BO44CC</div>
+          <span className={styles.name}>{user?.name} {user?.surname}</span>
```
Comment (ghost) [inline frontend/src/components/profile/components/profile-info/profile-info.tsx]:
```suggestion
          <span className={styles.name}>{user.name} {user?.surname}</span>
```
User should alway be on this page, so remove optional chaining

File: frontend/src/components/profile/profile.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import { RootState } from 'common/types';
+
+
+import styles from './styles.module.scss';
+
+const Profile: React.FC = () => {
+  const { user } = useSelector(({ auth }: RootState) => ({
+  user: auth.user
+  }));
+  const editUserInfo = () => {
```
Comment (ghost) [inline frontend/src/components/profile/profile.tsx]:
```suggestion
  const handleUserInfoEdit = () => {
```
use handle for local component cb

File: frontend/src/components/profile/styles.module.scss [hunk 7]
```diff
@@ -0,0 +1,9 @@
+.profileContainer {
+  display: flex;
+  height: 100vh;
```
Comment (ghost) [inline frontend/src/components/profile/styles.module.scss]:
```suggestion
  flex-grow: 1;
```

File: frontend/src/helpers/date/get-date-format.helper.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+
+const getDateFormat = (value:string):string => {
```
Comment (ghost) [inline frontend/src/helpers/date/get-date-format.helper.ts]: Also let's add https://day.js.org/ library for date formatting

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/components/profile/components/profile-info/profile-info.tsx [hunk 9]
```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import { IUser } from 'common/interfaces/user';
+import { DateFormat } from 'common/enums';
+import { getFormattedDate } from 'helpers';
+import clsx from "clsx";
+import styles from './styles.module.scss';
+
+type Props = {
+  user: IUser
+  edit: () => void
```
Comment (ghost) [inline frontend/src/components/profile/components/profile-info/profile-info.tsx]: Check **A1** criteria

Comment (ghost) [inline frontend/src/components/profile/components/profile-info/profile-info.tsx]: Create a type `type BindingCb = () => void` and put it to the shared folder(`src/common/types/cbs`)

File: frontend/src/components/profile/components/profile-info/profile-info.tsx [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+const ProfileInfo: React.FC<Props> = ({user, edit}) => {
+  const birthdate = getFormattedDate( user.birthdate, DateFormat.D_MMMM_YYYY);
+  return (
+    <div className={styles.mainInfo}>
+      <div className={styles.infoHeader}>
+        <span>My Profile</span>
+        <div className={styles.editButton} onClick={edit}>
```
Comment (ghost) [inline frontend/src/components/profile/components/profile-info/profile-info.tsx]: Use onlick only for button tag

File: frontend/src/components/profile/profile.tsx [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+import { IUser, IEditUserPayload } from 'common/interfaces';
+import { AuthActionCreator } from 'store/slices';
+import { useSelector } from 'react-redux';
+import { useDispatch } from 'react-redux';
+import { RootState } from 'common/types';
+
+import styles from './styles.module.scss';
+
+const Profile: React.FC = () => {
+  const [edite, setEdit] = useState(false);
```
Comment (ghost) [inline frontend/src/components/profile/profile.tsx]: Check **B6** criteria

Comment (ghost) [inline frontend/src/components/profile/profile.tsx]: Always types hooks manually

File: frontend/src/components/profile/profile.tsx [hunk 12]
```diff
@@ -0,0 +1,42 @@
+import React, { useState } from "react";
+import ProfileMenu from "./components/profile-menu/profile-menu";
+import ProfileInfo from "./components/profile-info/profile-info";
```
Comment (ghost) [inline frontend/src/components/profile/profile.tsx]:
```suggestion
import { ProfileMenu, ProfileInfo } from "./components";
```
Also check **A3** criteria

File: frontend/src/components/profile/profile.tsx [hunk 13]
```diff
@@ -0,0 +1,42 @@
+import React, { useState } from "react";
```
Comment (ghost) [inline frontend/src/components/profile/profile.tsx]:
```suggestion
import * as React from "react";
```

File: frontend/src/components/profile/profile.tsx [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+import ProfileMenu from "./components/profile-menu/profile-menu";
+import ProfileInfo from "./components/profile-info/profile-info";
+import { EditUserPopup } from 'components/admin-page/components';
```
Comment (ghost) [inline frontend/src/components/profile/profile.tsx]: if you reuse something - move it to the `common` components

File: frontend/src/components/profile/profile.tsx [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+
+import styles from './styles.module.scss';
+
+const Profile: React.FC = () => {
+  const [edite, setEdit] = useState(false);
+  const { user } = useSelector(({ auth }: RootState) => ({
+  user: auth.user as IUser
+  }));
+  const dispatch = useDispatch();
+  const openEditPopUp = () => {
```
Comment (ghost) [inline frontend/src/components/profile/profile.tsx]: Check **A9** criteria

File: frontend/src/store/slices/auth/auth.slice.ts [hunk 16]
```diff
@@ -48,6 +47,21 @@ const registration = createAsyncThunk(
   },
 );
 
+const editCurrentUser = createAsyncThunk(
+  'user/:id',
+  async (userData: IEditUserPayload) => {
+    try {
+      const user = await userApi.editUser((userData.id as string), userData);
+      return user[0];
```
Comment (ghost) [inline frontend/src/store/slices/auth/auth.slice.ts]: It shouldn't return an array

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/app.tsx [hunk 17]
```diff
@@ -4,13 +4,15 @@ import { AppRoute } from 'common/enums';
...
 import NotFound from 'components/not-found/not-found';
+import Profile from 'components/profile/profile';
 import Clinics from 'components/clinics/clinics';
 import DoctorsSearch from 'components/doctors-search/doctors-search';
 
 const App: React.FC = () => (
   <Switch>
     <Route path={AppRoute.ADMIN_PAGE} exact component={AdminPage}/>
     <Route path={[AppRoute.SIGN_IN, AppRoute.SIGN_UP]} component={Sign}/>
+    <Route path={AppRoute.MY_PROFILE} component={Profile} />
```
Comment (ghost) [inline frontend/src/app.tsx]:
```suggestion
    <Route path={AppRoute.USER_PROFILE} component={UserProfile} />
```
Rename component/file/folder and etc to `user profile`

**Review by ghost — CHANGES_REQUESTED** [review]

File: backend/src/services/user-service/user-service.service.ts [hunk 18]
```diff
@@ -11,6 +11,14 @@ class UserService {
     return userRepository.getByType(type);
   }
 
+  public async getUserById(id:string):Promise<IUser | null>{
+    const user = await userRepository.getById(id);
+    if (user && user.type === UserType.DOCTOR){
```
Comment (ghost) [inline backend/src/services/user-service/user-service.service.ts]:
```suggestion
    if (user?.type === UserType.DOCTOR){
```

File: frontend/src/components/common/edit-user/edit-user.tsx [hunk 19]
```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
```
Comment (ghost) [inline frontend/src/components/common/edit-user/edit-user.tsx]: Rename it to the `edit-user-popup`, wrap it into `Modal` component and pass `isOpen` prop from the parent

File: frontend/src/components/profile/components/documents/documents.tsx [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+  return (
+    <div className={styles.documentContainer}>
+      <span>Document status: </span><span>{document.status}</span>
+      <div className={styles.checkButton}>
+        <Button
+          label="Check document"
+          color={ButtonColor.GRAY_LIGHT}
+          styleType={ButtonStyleType.WITHOUT_BORDER}
+          hasHiddenLabel={false}
+          onClick={()=>setIsModalOpen(true)}
```
Comment (ghost) [inline frontend/src/components/profile/components/documents/documents.tsx]: Remove all anonymous from jsx

File: frontend/src/components/profile/components/user-info/user-info.tsx [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+const UserInfo: React.FC<Props> = ({ user, isCurrentUser, onEdit }) => {
+  const birthdate = getFormattedDate( user.birthdate, DateFormat.D_MMMM_YYYY);
+  return (
+    <div className={styles.mainInfo}>
+      <div className={styles.infoHeader}>
+        <span className={styles.title}>My Profile</span>
+        {isCurrentUser && <Button
+          label={''}
```
Comment (ghost) [inline frontend/src/components/profile/components/user-info/user-info.tsx]: Always give label

File: frontend/src/components/profile/components/user-info/user-info.tsx [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+import { BindingCb } from 'common/types';
+import { getFormattedDate } from 'helpers';
+import { Button } from 'components/common';
+import { ButtonColor, ButtonStyleType, ButtonIcon } from 'common/enums';
+import clsx from 'clsx';
+import styles from './styles.module.scss';
+
+type Props = {
+  user: IUser
+  isCurrentUser:boolean | undefined
```
Comment (ghost) [inline frontend/src/components/profile/components/user-info/user-info.tsx]:
```suggestion
  isCurrentUser:boolean
```

File: frontend/src/components/profile/profile.tsx [hunk 23]
```diff
@@ -0,0 +1,69 @@
...
+
+import styles from './styles.module.scss';
+
+type RouteParam = {
+  id:string
+};
+
+const Profile: React.FC = () => {
+  const [isEditeMode, setIsEditMode] = React.useState<boolean>(false);
+  const [user, setUser] = React.useState<UserGeneric>();
```
Comment (ghost) [inline frontend/src/components/profile/profile.tsx]:
```suggestion
  const [user, setUser] = React.useState<user>();
```
We need to create a new slice for it and don't share the logic with `user` slice. So remove all `isCurrent user` and other connected things with this

File: frontend/src/common/types/user/user-generic.type.ts [hunk 24]
```diff
@@ -0,0 +1,5 @@
+import { IUser, IUserTypeDoctor } from 'common/interfaces';
+
+type UserGeneric = IUser | IUserTypeDoctor | null;
```
Comment (ghost) [inline frontend/src/common/types/user/user-generic.type.ts]: We don't need to mix such things. Remove it

File: frontend/src/components/profile/profile.tsx [hunk 25]
```diff
@@ -0,0 +1,69 @@
...
+      {user ? <div className={styles.profileContainer}>
+        <SideMenu />
+        <div className={styles.infoContainer}>
+          <UserInfo user ={user} onEdit={handleOpenPopUp} isCurrentUser={isCurrentUser}/>
+          {isDoctor && <Documents document={(user as IUserTypeDoctor).doctor.document}/>}
+        </div>
+        {isEditeMode && <EditUserPopup
+          user={user}
+          onEditUser={handleUserInfoEdit}
+          onFormHide={() => setIsEditMode(false)}
```
Comment (ghost) [inline frontend/src/components/profile/profile.tsx]: Remove all `anonymous` functions from `jsx`

File: frontend/src/services/document-api/document-api.service.ts [hunk 26]
```diff
@@ -0,0 +1,33 @@
...
+class DocumentApi {
+  #http: Http;
+  #apiPrefix: string;
+
+  constructor({ http, apiPrefix }: Constructor) {
+    this.#http = http;
+    this.#apiPrefix = apiPrefix;
+  }
+
+  public editDocument(id: string, payload:any): Promise<IDocument> { //fix any latter
```
Comment (ghost) [inline frontend/src/services/document-api/document-api.service.ts]: Remove any

File: frontend/src/store/slices/users/users.slice.ts [hunk 27]
```diff
@@ -22,18 +24,24 @@ const { reducer, actions } = createSlice({
...
       state.users = state.users.filter(
         (user: IUser) => user.id !== action.payload,
       );
     },
+    setUserToProfile:(state, action: PayloadAction<UserGeneric>) => {
+      state.userInProfile = action.payload;
+    },
+    editeDocumentStatus:(state, action: PayloadAction<IDocument>) => {
+      (state.userInProfile as IUserTypeDoctor).doctor.document = action.payload;
+    },
```
Comment (ghost) [inline frontend/src/store/slices/users/users.slice.ts]: We don't need to share it with `users` slice. Create an own slice

**Review by ghost — CHANGES_REQUESTED** [review]
Good 👍

File: frontend/src/components/profile/components/documents/documents.tsx [hunk 28]
```diff
@@ -0,0 +1,92 @@
...
+import styles from './styles.module.scss';
+
+type Props = {
+  document: IDocument
+};
+
+const Status = {
+  [DocumentStatus.VERIFIED]: 'verified',
+  [DocumentStatus.IN_REVIEW]: 'in review',
+};
```
Comment (ghost) [inline frontend/src/components/profile/components/documents/documents.tsx]:
```suggestion
const documentStatusToReadable  = {
  [DocumentStatus.VERIFIED]: 'Verified',
  [DocumentStatus.IN_REVIEW]: 'In review',
};
```
Move it to the `frontend/src/components/profile/common/maps`

File: frontend/src/components/profile/components/documents/documents.tsx [hunk 29]
```diff
@@ -0,0 +1,92 @@
...
+
+const Documents: React.FC<Props> = ({ document }) => {
+  const [isModalOpen, setIsModalOpen] = React.useState<boolean>(false);
+  const [radioValue, setRadioValue] = React.useState<DocumentStatus>(document.status);
+  const dispatch = useDispatch();
+  const handleChangeRadio = (event:InputChangeEvent) => {
+    setRadioValue(event.target.value as DocumentStatus);
+  };
```
Comment (ghost) [inline frontend/src/components/profile/components/documents/documents.tsx]: We should use `useForm` for working with forms

File: frontend/src/components/profile/components/documents/documents.tsx [hunk 30]
```diff
@@ -0,0 +1,92 @@
...
+                options={[
+                  {
+                    label: Status[DocumentStatus.VERIFIED],
+                    value: DocumentStatus.VERIFIED,
+                  },
+                  {
+                    label: Status[DocumentStatus.IN_REVIEW],
+                    value: DocumentStatus.IN_REVIEW,
+                  },
+                ]}
```
Comment (ghost) [inline frontend/src/components/profile/components/documents/documents.tsx]:
```suggestion
                options={verifyDocumentOptions}
```
Create options outside of the component

File: frontend/src/components/profile/components/documents/documents.tsx [hunk 31]
```diff
@@ -0,0 +1,92 @@
...
+      <div className={styles.checkButton}>
+        <Button
+          label="Check document"
+          color={ButtonColor.GRAY_LIGHT}
+          styleType={ButtonStyleType.WITHOUT_BORDER}
+          hasHiddenLabel={false}
+          onClick={handleModalOpen}
+        />
+      </div>
+      <Modal isShow={isModalOpen}>
```
Comment (ghost) [inline frontend/src/components/profile/components/documents/documents.tsx]: Create new component for this full modal and move it to the `frontend/src/components/profile/components`

File: frontend/src/components/profile/profile.tsx [hunk 32]
```diff
@@ -0,0 +1,60 @@
...
+            <UserInfo user ={user} onEdit={handleTogglePopUp} />
+            {isDoctor && <Documents document={(user as IUserTypeDoctor).doctor.document}/>}
+          </div>
+          <Modal isShow={isEditeMode}>
+            <EditUserPopup
+              user={user}
+              onEditUser={handleUserInfoEdit}
+              onFormHide={handleTogglePopUp}
+            />
+          </Modal>
```
Comment (ghost) [inline frontend/src/components/profile/profile.tsx]: Wrap `EditUserPopup` with `Modal` inside `EditUserPopup` and pass the props `isOpen from the parent`

**Review by ghost — CHANGES_REQUESTED** [review]
👍

File: frontend/src/components/common/radio/radio.tsx [hunk 33]
```diff
@@ -1,10 +1,12 @@
...
 import clsx from 'clsx';
 import { IOption } from 'common/interfaces';
 import { InputChangeCallback } from 'common/types';
-import * as React from 'react';
+import { UseFormMethods } from 'react-hook-form';
+
 import styles from './styles.module.scss';
 
-interface Props {
+interface Props extends Partial<Pick<UseFormMethods, 'register'>>{
```
Comment (ghost) [inline frontend/src/components/common/radio/radio.tsx]: We already have this type. Check `checkbox` component

File: frontend/src/components/common/radio/radio.tsx [hunk 34]
```diff
@@ -13,31 +15,34 @@ interface Props {
   onChange: InputChangeCallback;
```
Comment (ghost) [inline frontend/src/components/common/radio/radio.tsx]:
```suggestion
```
We don't need it if we use `useForm`;

Delete `InputChangeCallback ` is we don't use it

Add `errors` props 

Remove `hasError` prop

File: frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx [hunk 35]
```diff
@@ -0,0 +1,74 @@
...
+];
+
+const EditeDocumentPopup: React.FC<Props> = ({ document, isShow, onToggleModal, onEditeDocument }) => {
+  const [radioValue, setRadioValue] = React.useState<DocumentStatus>(document.status);
+  const { register, handleSubmit, watch } = useForm<DocumentFormData>();
+
+  const handleChangeRadio = () => {
+    const { status } = watch();
+    setRadioValue(status);
+  };
```
Comment (ghost) [inline frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx]:
```suggestion
```
We don't need it if we use use form

File: frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx [hunk 36]
```diff
@@ -0,0 +1,74 @@
...
+
+const verifyDocumentOptions = [
+  {
+    label: documentStatusToReadable[DocumentStatus.VERIFIED],
+    value: DocumentStatus.VERIFIED,
+  },
+  {
+    label: documentStatusToReadable[DocumentStatus.IN_REVIEW],
+    value: DocumentStatus.IN_REVIEW,
+  },
```
Comment (ghost) [inline frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx]: Use `createOption` here

```js
const verifyDocumentOptions = createOption<string>(Object.values(DocumentStatus), (documentStatus) => ({
  value: documentStatus,
  label: documentStatusToReadable[DocumentStatus.VERIFIED],
}))
```

File: frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx [hunk 37]
```diff
@@ -0,0 +1,74 @@
...
+  };
+  const handleSubmitForm = (formData:DocumentFormData) => {
+    const update:IDocument = { ...document, status: formData.status };
+    onEditeDocument(update);
+  };
+  return (
+    <Modal isShow={isShow}>
+      <div className={styles.formContainer}>
+        <button className={styles.closeButton} onClick={onToggleModal} type="button">
+          &#10060;
```
Comment (ghost) [inline frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx:44]: Add correct label
use `visually-hidden` class, if we don't have a label in the markup. See **A12**

File: frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx [hunk 38]
```diff
@@ -0,0 +1,74 @@
...
+        </button>
+        <form className={styles.form} onSubmit={handleSubmit(handleSubmitForm)}>
+          <div className={styles.radio}>
+            <Radio
+              options={verifyDocumentOptions}
+              register={register}
+              value={radioValue}
+              name="status"
```
Comment (ghost) [inline frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx]: Use Status key enum here

**Review by ghost — CHANGES_REQUESTED** [review]
Good 👍

File: frontend/src/components/common/radio/radio.tsx [hunk 39]
```diff
@@ -1,43 +1,46 @@
...
-            value={option.value}
-            className={styles.radioInput}
-            type="radio"
-          />
-        </label>
-      ))
-    }
-  </div>
-);
+const Radio: React.FC<Props> = ({ options, register, value, isDisabled, errors, name }) => {
```
Comment (ghost) [inline frontend/src/components/common/radio/radio.tsx:18]: We need to output error with `ErrorMessage`. See `checkbox` component

File: frontend/src/components/profile/common/enums/document-form-key.tsx [hunk 40]
```diff
@@ -0,0 +1,5 @@
+enum DocumentFormKey {
```
Comment (ZubkoWhatislove) [inline frontend/src/components/profile/common/enums/document-form-key.tsx:1]: We don't need it. Use `DocumentKey` enum that we already have

Comment (ZubkoWhatislove) [reply frontend/src/components/profile/common/enums/document-form-key.tsx:1]: Remove it

File: frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx (hunk 37 above)
Comment (ghost) [reply frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx:44]: Missed

We need to do something like
```html
<button>
  &#10060;
  <span className="visually-hidden">Open edit document popup</span>
</button>
```
Check **A12** criteria

File: frontend/src/components/profile/common/types/document-form-data.tsx [hunk 41]
```diff
@@ -0,0 +1,8 @@
+import { DocumentStatus } from 'common/enums';
```
Comment (ghost) [inline frontend/src/components/profile/common/types/document-form-data.tsx]: It isn't `tsx` file

File: shared/src/common/interfaces/edite-document/edit-document-payload.interface.ts [hunk 42]
```diff
@@ -0,0 +1,11 @@
+import { EditDocumentPayloadKey, DocumentStatus } from '~/common/enums';
+
+type IEditDocumentPayload = {
```
Comment (ghost) [inline shared/src/common/interfaces/edite-document/edit-document-payload.interface.ts]: `I` can't be type

Comment (ZubkoWhatislove) [inline shared/src/common/interfaces/edite-document/edit-document-payload.interface.ts:3]: This is the same as IDocument, isn't it?
Remove it

File: frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx [hunk 43]
```diff
@@ -0,0 +1,73 @@
...
+  });
+  const status = watch(DocumentFormKey.STATUS, document.status);
+
+  const handleSubmitForm = (formData:DocumentFormData) => {
+    const update = { ...document, status: formData.status };
+    onEditeDocument(update);
+  };
+  return (
+    <Modal isShow={isShow}>
+      <div className={styles.visualyHidden}>
```
Comment (ghost) [inline frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx]: Rename visually-hidden

File: frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx [hunk 44]
```diff
@@ -0,0 +1,73 @@
...
+        <button className={styles.closeButton} onClick={onToggleModal} type="button">
+          &#10060;
+        </button>
+        <form className={styles.form} onSubmit={handleSubmit(handleSubmitForm)}>
+          <div className={styles.radio}>
+            <Radio
+              options={verifyDocumentOptions}
+              register={register}
+              value={status}
+              name={DocumentFormKey.STATUS}
```
Comment (ZubkoWhatislove) [inline frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx:51]:
```suggestion
              name={DocumentKey.DOCUMENT_STATUS}
```

File: frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx [hunk 45]
```diff
@@ -0,0 +1,73 @@
...
+  onEditeDocument:EditeDocumentCb;
+};
+
+const verifyDocumentOptions = createOptions<string>(Object.values(DocumentStatus), (documentStatus) => ({
+  value: documentStatus,
+  label: documentStatusToReadable[(documentStatus as DocumentStatus)],
+}));
+
+const EditeDocumentPopup: React.FC<Props> = ({ document, isShow, onToggleModal, onEditeDocument }) => {
+  const defaultValues = { [DocumentFormKey.STATUS]: document.status };
```
Comment (ghost) [inline frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx]:
```suggestion
  const DEFAULT_VALUES = {
    [DocumentFormKey.STATUS]: document.status,
  };
```
Move it to the `frontend/src/components/profile/components/edite-document-popup/common/constants`

File: frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx [hunk 46]
```diff
@@ -0,0 +1,73 @@
...
+  const { register, handleSubmit, watch, errors } = useForm<DocumentFormData>({
+    defaultValues: defaultValues,
+  });
+  const status = watch(DocumentFormKey.STATUS, document.status);
+
+  const handleSubmitForm = (formData:DocumentFormData) => {
+    const update = { ...document, status: formData.status };
```
Comment (ghost) [inline frontend/src/components/profile/components/edite-document-popup/edite-document-popup.tsx]:
```suggestion
    const updatedDocument: IDocument = {
      ...document,
      status: formData.status
    };
```

**Review by ghost — CHANGES_REQUESTED** [review]
Change the IDocument
This interface has `DocumentStatus` key but in the DB we have only `status`

File: shared/src/common/interfaces/edite-document/edit-document-payload.interface.ts [hunk 47]
```diff
@@ -0,0 +1,11 @@
...
+  [EditDocumentPayloadKey.ID]?: string;
```
Comment (ghost) [inline shared/src/common/interfaces/edite-document/edit-document-payload.interface.ts]:
```suggestion
  [EditDocumentPayloadKey.ID]: string;
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: shared/src/common/interfaces/edite-document/edit-document-payload.interface.ts [hunk 48]
```diff
@@ -0,0 +1,11 @@
...
+  [EditDocumentPayloadKey.IMAGE_PATH]: string;
+  [EditDocumentPayloadKey.DOCUMENT_STATUS]: DocumentStatus;
```
Comment (ghost) [inline shared/src/common/interfaces/edite-document/edit-document-payload.interface.ts]:
```suggestion
  [EditDocumentPayloadKey.STATUS]: DocumentStatus;
```

File: shared/src/common/enums/edite-document/edit-document-payload.enums.ts [hunk 49]
```diff
@@ -0,0 +1,9 @@
+enum EditDocumentPayloadKey {
+  ID = 'id',
+  IMAGE_PATH = 'imagePath',
+  DOCUMENT_STATUS = 'status',
```
Comment (ghost) [inline shared/src/common/enums/edite-document/edit-document-payload.enums.ts]:
```suggestion
  STATUS = 'status',
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]
👍

**Pull request merged**